swap/src/bitcoin/wallet.rs

```diff
@@ -301,11 +305,30 @@ where
         let wallet = self.wallet.lock().await;
         let client = self.client.lock().await;
         let fee_rate = client.estimate_feerate(self.target_block)?;
+        let script = address.script_pubkey();
 
         let mut tx_builder = wallet.build_tx();
-        tx_builder.add_recipient(address.script_pubkey(), amount.as_sat());
+        tx_builder.add_recipient(script.clone(), amount.as_sat());
         tx_builder.fee_rate(fee_rate);
         let (psbt, _details) = tx_builder.finish()?;
+        let mut psbt: PartiallySignedTransaction = psbt;
+
+        // When subscribing to transactions we depend on the relevant script being at
+        // output index 0, thus we ensure the relevant output to be at index `0`.
+        psbt.outputs.sort_by(|a, _| {
+            if a.witness_script == Some(script.clone()) {
+                Ordering::Less
+            } else {
+                Ordering::Greater
+            }
+        });
+        psbt.global.unsigned_tx.output.sort_by(|a, _| {
+            if a.script_pubkey == script {
+                Ordering::Less
+            } else {
+                Ordering::Greater
+            }
```

Comment:
I thought I just commented but it seem to have disappeared? 

If I understand you correctly than we won't find the lock transaction later on if the locking script is not on vout 0. 

If this is true: what do you think of asserting here after the sort and fail the function if the script is not on vout 0?

Reply:
We will always find the transaction, just subscribing to a different script but for checking confirmations, that doesn't matter.

This sorting only fixes that we can reuse a previous (program internal) subscription and don't check twice for the same TX.